trino-s3-proxy/src/test/java/io/trino/s3/proxy/server/testing/TestingTrinoS3ProxyServer.java

```diff
@@ -14,37 +14,73 @@
 package io.trino.s3.proxy.server.testing;
 
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
 import com.google.inject.Injector;
 import com.google.inject.Module;
 import io.airlift.bootstrap.Bootstrap;
+import io.airlift.bootstrap.LifeCycleManager;
 import io.airlift.event.client.EventModule;
-import io.airlift.http.server.testing.TestingHttpServer;
 import io.airlift.http.server.testing.TestingHttpServerModule;
 import io.airlift.jaxrs.JaxrsModule;
 import io.airlift.json.JsonModule;
-import io.airlift.log.Logger;
 import io.airlift.node.testing.TestingNodeModule;
 import io.trino.s3.proxy.server.credentials.Credentials;
-import io.trino.s3.proxy.server.credentials.Credentials.Credential;
+
+import java.io.Closeable;
 
 public final class TestingTrinoS3ProxyServer
+        implements Closeable
 {
-    private static final Logger log = Logger.get(TestingTrinoS3ProxyServer.class);
+    private final Injector injector;
+
+    private TestingTrinoS3ProxyServer(Injector injector)
+    {
+        this.injector = injector;
+    }
+
+    @Override
+    public void close()
+    {
+        injector.getInstance(LifeCycleManager.class).stop();
+    }
+
+    public Injector getInjector()
+    {
+        return injector;
+    }
 
-    private TestingTrinoS3ProxyServer() {}
+    public static Builder builder()
+    {
+        return new Builder();
+    }
 
-    public static void main(String[] args)
+    public static class Builder
     {
-        if (args.length != 4) {
-            System.err.println("Usage: TestingTrinoS3ProxyServer <emulatedAccessKey> <emulatedSecretKey> <realAccessKey> <realSecretKey>");
-            System.exit(1);
+        private final ImmutableSet.Builder<Credentials> credentials = ImmutableSet.builder();
+
+        private Builder()
+        {
+        }
```

Comment:
nit: this all can be on one line:
```suggestion
private Builder() {}
```